app/models/concerns/host_common.rb

```diff
@@ -15,14 +15,14 @@ module HostCommon
   included do
     belongs_to_proxy :puppet_proxy,
       :feature => N_('Puppet'),
-      :label => N_('Puppet Master'),
-      :description => N_('Use this puppet server as an initial Puppet Server or to execute puppet runs'),
+      :label => N_('Puppetserver Proxy'),
+      :description => N_('Use this Smart Proxy to connect to an initial Puppetserver to execute puppet runs'),
```

Comment:
I'm not sure the initial part is correct. It's also presented in the ENC

Reply:
So the value is specifically used by default during provision - correct? which I'm guessing is where the "initial" originally came from. It doesn't force the use of the defined CA proxy / Puppet Proxy either. hm

"Smart Proxy provided to provisioning templates and ENC as the Puppet Server" 

... that sounds awkward... I'll drink some more coffee and keep thinking

Reply:
Yes, this is hard stuff. Perhaps @melcorr has some good thoughts.

IMHO there are 2 things that are relevant. First of all, there's just the intention. Whenever you want to know which server is responsible, this is the authoritative value. Like a CMDB. Other tools and integrations can read this. One of those would be provisioning, but the ENC also sends it so you can correct it. Ansible / REX could also use it to update. The other side is importing classes. That calls the Smart Proxy which in turn calls the Puppetserver API.

Now to summarize that in a short description ...

Reply:
I might be totally off but it might get you thinking:

`Enter the Smart Proxy to use for provisioning templates and as the Puppet Server ENC`

or 

`Enter the Smart Proxy to use as both the default Puppet Server for Puppet runs and the Puppet Server ENC`

Reply:
`Enter the default Smart Proxy used to access the Puppetserver API`

and

`Enter the default Smart Proxy used to access the Puppetserver CA API`

Reply:
See the latest - let me know if you want me to tweak it anymore - This at least causes me to think/look more and not assume the value I'm picking there is the Server that I want to use, but rather the Smart Proxy... which would when troubleshooting trigger me to look at the Smart Proxy for issues if values aren't correct... So I think this accomplishes the original goal.